lib/taurus/core/taurusmanager.py

```diff
@@ -108,27 +111,49 @@ def cleanUp(self):
 
         self._thread_pool.join()
         self._thread_pool = None
+        self._sthread_pool.join()
+        self._sthread_pool = None
 
         self._state = ManagerState.CLEANED
 
     def addJob(self, job, callback=None, *args, **kw):
-        """Add a new job (callable) to the queue. The new job will be processed
-        by a separate thread
+        """ Deprecated. Wrapper of enqueueJob. See enqueueJob documentation.
+        """
+        self.deprecated(dep='addJob', alt='enqueueJob', rel='4.3.2')
+        self.enqueueJob(job, callback=callback, jobargs=args, jobkwargs=kw)
 
+    def enqueueJob(self, job, callback=None, jobargs=None, jobkwargs=None,
+                   serialization_mode=None):
+        """ Enqueue a job (callable) to the queue. The new job will be
+        processed by a separate thread
         :param job: (callable) a callable object
         :param callback: (callable) called after the job has been processed
-        :param args: (list) list of arguments passed to the job
-        :param kw: (dict) keyword arguments passed to the job
+        :param jobargs: (list) list of arguments passed to the job
+        :param jobkwargs: (dict) keyword arguments passed to the job
+        :param serialization_mode:
+        (taurus.core.taurusbasetypes.TaurusSerializationMode) serialization
+        mode
         """
-        if self._serialization_mode == TaurusSerializationMode.Concurrent:
+        if serialization_mode is None:
+            serialization_mode = self._serialization_mode
+
+        if serialization_mode == TaurusSerializationMode.Concurrent:
             if not hasattr(self, "_thread_pool") or self._thread_pool is None:
                 self.info("Job cannot be processed.")
                 self.debug(
-                    "The requested job cannot be processed. Make sure this manager is initialized")
+                    "The requested job cannot be processed. " +
+                    "Make sure this manager is initialized")
                 return
-            self._thread_pool.add(job, callback, *args, **kw)
+            self._thread_pool.add(job, callback, jobargs, jobkwargs)
```

Comment:
I think that it should be `self._thread_pool.add(job, callback, *jobargs, **jobkwargs)`

Reply:
... and then thre should be another elif for supporting "TangoSerial" and finally an `else` for raising a `RuntimeError("Unsupported serialization mode {}".format(serialization_mode))`

Reply:
> I think that it should be self._thread_pool.add(job, callback, *jobargs, **jobkwargs)
This is wrong but IMO it should be `self._thread_pool.add(job, callback, args=jobargs, kwargs=jobkwargs)`

See:

```
def foo(a,b, *args, **kwargs):
    print "foo", args, kwargs


def newfoo(job, callback, jobargs=None, jobkwargs=None, myfoo=None):
    # foo(job, callback, *jobargs, **jobkwargs) # TypeError: foo() argument after ** must be a mapping, not NoneType
    foo(job, callback, args=jobargs, kwargs=jobkwargs)


newfoo(1,2)
newfoo(1,2, (3, 4, 5), {'1': 1})
```

Reply:
Please have a look at the signature of `taurus.core.util.threadpool.ThreadPool.add()`

It is: `add(self, job, callback=None, *args, **kw)`

This means that the args and kwargs are expected "expanded" ... and we do not want to change that API now.